File: prog6.py
year=int(input("Enter the year: "))

if year%4==0:
    if year%100==0 and (year//100)%4!=0:
        print(f'{year} is not a leap year')
    elif year%400==0:
        print(f'{year} is a leap year')
    elif year%100!=0:
        print(f'{year} is a leap year')
else:
    print(f'{year} is not a leap year')

File: prog6.c
#include<stdio.h>

int main()
{
    int year;
    printf("Enter the year: ");
    scanf("%i",&year);

    if (year%400==0)
    {
        printf("The year %i is a leap year",year);
    }
    else 
    {
        if (year%100==0)
        {
            printf("The year %i is not a leap year",year);
        }
        else if (year%4==0)
        {
            printf("The year %i is a leap year.",year);
        }
        else
        {
            printf("The year %i is not a leap year",year);
        }
    }
    
    return 0;
}